app/src/main/java/com/example/instagram/PostsAdapter.java

```diff
@@ -61,14 +63,24 @@ public void clear() {
     class ViewHolder extends RecyclerView.ViewHolder {
         private TextView tvUsername;
         private ImageView ivImage;
-        final View rootView;
+        private ImageView ivLikeButton;
+        View rootView;
+        private Post post;
         private TextView tvDescription;
+        private boolean liked;
+
         public ViewHolder(@NonNull View itemView) {
             super(itemView);
             tvUsername = itemView.findViewById(R.id.tvUsername);
             ivImage = itemView.findViewById(R.id.ivImage);
             tvDescription = itemView.findViewById(R.id.tvDetailedDescription);
+            ivLikeButton = itemView.findViewById(R.id.ivLikeButton);
             rootView = itemView;
+            post = (Post) rootView.getTag();
+
+            if (post.)
```

Comment:
Going forward, let's not commit code that would cause your project to fail to build like this line.